jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/RunCommandDialog.kt

```diff
@@ -27,7 +33,9 @@ import javax.swing.JComponent
 import javax.swing.JTextField
 import javax.swing.plaf.basic.BasicComboBoxEditor
 
-class RunCommandDialog(private val project: Project, private val container: ContainerDetails) : DialogWrapper(project) {
+class RunCommandDialog(private val project: Project, private val container: ContainerDetails) :
+    DialogWrapper(project),
+    CoroutineScope by ApplicationThreadPoolScope("RunCommandDialog") {
```

Comment:
i thought we were moving away from coroutine scope by delegation?

Reply:
We should be.

`private val coroutineScope = ApplicationThreadPoolScope("RunCommandDialog", disposable)` would be the more correct route.